contribs/github-bot/conditions/file.go

```diff
@@ -0,0 +1,58 @@
+package conditions
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/gnolang/gno/contribs/github-bot/client"
+	"github.com/gnolang/gno/contribs/github-bot/utils"
+
+	"github.com/google/go-github/v64/github"
+	"github.com/xlab/treeprint"
+)
+
+// FileChanged Condition
+type fileChanged struct {
+	gh      *client.GitHub
+	pattern *regexp.Regexp
+}
+
+var _ Condition = &fileChanged{}
+
+func (fc *fileChanged) IsMet(pr *github.PullRequest, details treeprint.Tree) bool {
+	detail := fmt.Sprintf("A changed file match this pattern: %s", fc.pattern.String())
+	opts := &github.ListOptions{
+		PerPage: client.PageSize,
+	}
+
+	for {
+		files, response, err := fc.gh.Client.PullRequests.ListFiles(
```

Comment:
I need to understand who thought this HTTP request pattern for pagination was a good idea in Go :)

Reply:
Yep and with go 1.23 you can now use iterators FYI ;)
https://github.com/google/go-github?tab=readme-ov-file#iterators-experimental